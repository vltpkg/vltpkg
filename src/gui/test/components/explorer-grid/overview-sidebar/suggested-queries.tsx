import { test, expect, vi, beforeEach, afterEach } from 'vitest'
import {
  waitFor,
  cleanup,
  fireEvent,
  render,
} from '@testing-library/react'
import html from 'diffable-html'
import { useGraphStore as useStore } from '@/state/index.ts'
import {
  SuggestedQueries,
  queries,
} from '@/components/explorer-grid/overview-sidebar/suggested-queries.tsx'

vi.mock('lucide-react', () => ({
  ChevronRight: 'gui-chevron-right',
  Search: 'gui-search',
  CornerDownRight: 'gui-corner-down-right',
}))

vi.mock('@/components/ui/button.tsx', () => ({
  Button: 'gui-button',
}))

vi.mock('@/components/query-bar/query-token.tsx', () => ({
  QueryToken: 'gui-query-token',
}))

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

beforeEach(() => {
  localStorage.setItem('suggestedQueriesExpanded', 'true')
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
  localStorage.clear()
})

test('suggested-queries renders default', () => {
  const Container = () => {
    return <SuggestedQueries />
  }

  const { container } = render(<Container />)
  expect(container.innerHTML).toMatchSnapshot()
})

test.each(queries)('suggested-queries renders query %s', async q => {
  useStore.setState({ query: ':root' })

  const Container = () => {
    return <SuggestedQueries />
  }

  const { getByTestId } = render(<Container />)
  const section = getByTestId(q.name)
  const button = section.querySelector('[role="button"]')!
  fireEvent.click(button)

  await waitFor(() => {
    const updatedQuery = useStore.getState().query
    expect(updatedQuery).toBe(`:root ${q.query}`)
  })
})
