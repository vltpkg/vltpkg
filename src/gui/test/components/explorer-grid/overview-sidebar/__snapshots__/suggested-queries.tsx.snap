// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`suggested-queries renders default 1`] = `

<div class="my-4 flex cursor-default flex-col gap-2 overflow-hidden rounded-lg border-[1px] border-dashed border-muted bg-muted-foreground/5 p-3">
  <div class="flex w-full items-center justify-between">
    <p class="text-sm font-medium tracking-tight">
      Suggested Queries
    </p>
    <gui-button
      size="icon"
      classname="size-6 rounded-sm"
      variant="secondary"
    >
      <span>
        <gui-chevron-right classname="text-muted-foreground">
        </gui-chevron-right>
      </span>
    </gui-button>
  </div>
  <div
    class="flex flex-col divide-y-[1px] divide-muted"
    style="height: auto; filter: blur(0px); opacity: 1;"
  >
    <div
      data-testid="Outdated"
      class="flex flex-col gap-1 py-3"
    >
      <p class="text-sm font-medium text-foreground">
        Outdated
      </p>
      <p class="text-xs text-muted-foreground">
        Find packages that are outdated
      </p>
      <div
        role="button"
        class="duration-250 group mt-1 flex cursor-default items-center gap-2 rounded-sm border-[1px] border-muted bg-background px-2 py-1.5 transition-colors hover:border-muted-foreground/50"
      >
        <gui-search
          size="16"
          classname="duration-250 text-muted-foreground transition-colors group-hover:text-foreground"
        >
        </gui-search>
        <div class="flex">
          <gui-query-token variant="pseudo">
            :outdated(
          </gui-query-token>
          <gui-query-token variant="tag">
            major
          </gui-query-token>
          <gui-query-token variant="pseudo">
            )
          </gui-query-token>
        </div>
        <gui-corner-down-right
          size="16"
          classname="duration-250 ml-auto text-muted-foreground/40 transition-colors group-hover:text-foreground"
        >
        </gui-corner-down-right>
      </div>
    </div>
    <div
      data-testid="Unmaintained"
      class="flex flex-col gap-1 py-3"
    >
      <p class="text-sm font-medium text-foreground">
        Unmaintained
      </p>
      <p class="text-xs text-muted-foreground">
        Find packages that are unmaintained
      </p>
      <div
        role="button"
        class="duration-250 group mt-1 flex cursor-default items-center gap-2 rounded-sm border-[1px] border-muted bg-background px-2 py-1.5 transition-colors hover:border-muted-foreground/50"
      >
        <gui-search
          size="16"
          classname="duration-250 text-muted-foreground transition-colors group-hover:text-foreground"
        >
        </gui-search>
        <div class="flex">
          <gui-query-token variant="pseudo">
            :unmaintained
          </gui-query-token>
        </div>
        <gui-corner-down-right
          size="16"
          classname="duration-250 ml-auto text-muted-foreground/40 transition-colors group-hover:text-foreground"
        >
        </gui-corner-down-right>
      </div>
    </div>
    <div
      data-testid="System access"
      class="flex flex-col gap-1 py-3"
    >
      <p class="text-sm font-medium text-foreground">
        System access
      </p>
      <p class="text-xs text-muted-foreground">
        Find packages with access to filesystem or environment variables
      </p>
      <div
        role="button"
        class="duration-250 group mt-1 flex cursor-default items-center gap-2 rounded-sm border-[1px] border-muted bg-background px-2 py-1.5 transition-colors hover:border-muted-foreground/50"
      >
        <gui-search
          size="16"
          classname="duration-250 text-muted-foreground transition-colors group-hover:text-foreground"
        >
        </gui-search>
        <div class="flex">
          <gui-query-token variant="pseudo">
            :is(
          </gui-query-token>
          <gui-query-token variant="pseudo">
            :fs
          </gui-query-token>
          <gui-query-token variant="selector">
            ,
          </gui-query-token>
          <gui-query-token variant="pseudo">
            :env
          </gui-query-token>
          <gui-query-token variant="pseudo">
            )
          </gui-query-token>
        </div>
        <gui-corner-down-right
          size="16"
          classname="duration-250 ml-auto text-muted-foreground/40 transition-colors group-hover:text-foreground"
        >
        </gui-corner-down-right>
      </div>
    </div>
  </div>
</div>

`;
